src/dbt_client/exception.ts

```diff
@@ -0,0 +1,40 @@
+import { PythonException } from "python-bridge";
+
+export class CustomException {
+  constructor(
+    public name: string,
+    public message: string,
+    public error?: Error,
+    public extra?: any,
+  ) {}
+
+  getErrorMessage() {
+    if (this.error) {
+      return this.error.stack || `${this.error.name}:${this.error.message}`;
+    }
+    return this.message;
+  }
+}
+
+export class CustomPythonException extends CustomException {
+  constructor(
+    public name: string,
+    message: string,
+    public error: PythonException,
+    public extra?: any,
+  ) {
+    super(name, message, error, extra);
+    this.message = `${message} : ${error.exception.message}`;
+  }
+}
+
+export class CustomUnknownException extends CustomException {
+  constructor(
+    public name: string,
+    message: string,
+    public extra: any,
+  ) {
+    super(name, message, undefined, extra);
+    this.message = `${message} : ${extra}`;
```

Comment:
if `extra` is a json object then this will results in `some message : [object Object]`. You may have to stringify or should not include in message